buildSrc/src/main/kotlin/JsPlugin.kt

```diff
@@ -0,0 +1,104 @@
+        configureJsCompilation(target)
+        configureJsTest(target)
+    }
+
+    private fun configureJsCompilation(target: Project) {
+        target.extensions.configure(KotlinMultiplatformExtension::class.java) {
+            targetFromPreset(presets.getByName("js"), "js")
+            sourceSets.getByName("jsMain").dependencies {
+                implementation(kotlin("stdlib-js"))
+            }
+            sourceSets.getByName("jsTest").dependencies {
+                implementation(kotlin("test-js"))
+            }
+        }
+        target.tasks.named("compileKotlinJs", Kotlin2JsCompile::class.java) {
+            kotlinOptions.configure()
+        }
+        target.tasks.named("compileTestKotlinJs", Kotlin2JsCompile::class.java) {
+            kotlinOptions.configure()
+        }
+    }
+
+    private fun configureJsTest(target: Project) {
+        // workaround for https://github.com/srs/gradle-node-plugin/issues/301
+        target.repositories.whenObjectAdded {
+            if (this is IvyArtifactRepository) {
+                metadataSources {
+                    artifact()
+                }
+            }
+        }
+
+        target.pluginManager.apply(NodePlugin::class.java)
+        target.extensions.configure(NodeExtension::class.java) {
+            download = true
+            version = "12.2.0"
+        }
+
+        val dependenciesTaskProvider = target.tasks.register("installJsTestDependencies", NpmTask::class.java) {
+            setArgs(listOf("install", "kotlin@1.3.31", "kotlin-test@1.3.31", "mocha@6.1.4"))
+        }
+
+        val nodeModuleTaskProvider = target.tasks.register("populateNodeModule", PopulateNodeModuleTask::class.java) {
+            input.set(target.tasks.named("compileKotlinJs", Kotlin2JsCompile::class.java).map { it.outputFile })
+        }
+
+        val compileTestJsTask = target.tasks.named("compileTestKotlinJs", Kotlin2JsCompile::class.java)
+
+        target.tasks.register("runMocha", NodeTask::class.java) {
+            dependsOn(dependenciesTaskProvider, nodeModuleTaskProvider, compileTestJsTask)
+            setScript(project.file("node_modules/mocha/bin/mocha"))
+            // NodeTask does not support lazy configuration through properties
+            setArgs(listOf(compileTestJsTask.get().outputFile.relativeTo(project.projectDir)))
+        }
+    }
+
+    private fun KotlinJsOptions.configure() {
+        metaInfo = true
+        sourceMap = true
+        sourceMapEmbedSources = "always"
+        moduleKind = "umd"
+        main = "call"
+    }
+
+    abstract class PopulateNodeModuleTask : AbstractTask() {
+
+        @get:InputFile
+        abstract val input: Property<File>
+
+        @get:OutputFile
+        abstract val output: Property<File>
+
+        init {
+            output.set(input.map { project.file("node_modules/${it.name}") })
+        }
+
+        @TaskAction
+        open fun populate() {
+            input.get().copyTo(target = output.get(), overwrite = true)
```

Comment:
Can we use [Copy](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.Copy.html) task for this?